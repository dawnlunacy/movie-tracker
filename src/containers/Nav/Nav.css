.nav {
  align-items: center;
  display: flex;
  justify-content: center;
}

.nav-p {
  margin: 0px 50px 0px 50px;
}
