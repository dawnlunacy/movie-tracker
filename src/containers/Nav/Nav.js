import React from 'react';
import { Route, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { logoutUser } from '../../actions/index';
import './Nav.css'

export const Nav = ({ currentUser, logoutUser, getFavorites }) => {
    return(
        <nav className="nav">
            {currentUser !== null && <Link className="link" to='/favorites'> <p className="nav-p" onClick={() => console.log('pooop-->',getFavorites())}>FAVORITES</p> </Link>}
            {currentUser === null && <p className="nav-p">LOGIN TO VIEW FAVORITES</p> }
            <div className="nav-user-options">
            {currentUser !== null && <p className="nav-p">Hello, {currentUser.name}</p>}
            {currentUser !== null && <Link className="link" to='/'> <p className="nav-p" onClick={() => logoutUser(currentUser)}>LOGOUT</p> </Link>}
            {currentUser === null && <Link className="link" to='/login'> <p className="nav-p">LOGIN</p> </Link>}
            {currentUser === null && <Link className="link" to='/signup'> <p className="nav-p">SIGN UP</p> </Link>}
            </div>
        </nav>
    )
}

export const mapStateToProps = state => ({
  currentUser: state.currentUser
});

export const mapDispatchToProps = dispatch => ({
  logoutUser: currentUser => dispatch(logoutUser(currentUser))
})

export default connect(mapStateToProps, mapDispatchToProps)(Nav);
