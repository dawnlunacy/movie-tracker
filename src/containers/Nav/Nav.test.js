import React from 'react';
import Nav from './Nav';
import { shallow } from 'enzyme';

describe('Nav', () => {

  it('should match snapshot', () => {
    const wrapper = shallow(<Nav />)

    expect(wrapper).toMatchSnapshot();
  })

})
