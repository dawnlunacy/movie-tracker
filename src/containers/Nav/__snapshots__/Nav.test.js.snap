// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
<nav
  className="nav"
>
  <p
    className="nav-p"
  >
    FAVORITES
  </p>
  <div
    className="nav-user-options"
  >
    <Link
      className="link"
      to="/login"
    >
       
      <p
        className="nav-p"
      >
        LOGIN
      </p>
       
    </Link>
    <Link
      className="link"
      to="/signup"
    >
       
      <p
        className="nav-p"
      >
        SIGN UP
      </p>
       
    </Link>
  </div>
</nav>
`;
